hera_cal/abscal.py

```diff
@@ -668,7 +669,49 @@ def delay_slope_lincal(model, data, antpos, wgts=None, refant=None, df=9.765625e
     return fit
 
 
-def global_phase_slope_logcal(model, data, antpos, wgts=None, refant=None, verbose=True, tol=1.0, edge_cut=0):
+def dft_phase_slope_solver(xs, ys, data):
+    '''Solve for sptial phase slopes across an array by looking for the peak in the DFT.
+    This is analogous to the method in utils.fft_dly(), except its in 2D and does not 
+    assume a regular grid for xs and ys.
+
+    Arguments:
+        xs: 1D array of x positions (e.g. of antennas or baselines)
+        ys: 1D array of y positions (must be same length as xs)
+        data: ndarray of complex numbers to fit with a phase slope. The first dimension must match 
+            xs and ys, but subsequent dimensions will be preserved and solved independently. 
+            Any np.nan in data is interpreted as a flag.
```

Comment:
Are we requiring all functions to be able to handle nans?  Or should the user be required to provide non-nan inputs?

Reply:
Added the ability for this function to take boolean flags.